### ISSUE TITLE
About eval while tuning

### ISSUE THREAD

**Issue by Snowilv** [RUCAIBox/LC-Rec#2]
Hi, thanks for the impressive work!

how could we evaluate LLM on the eval set while tuning?  I  can't find related codes in the repo

**Comment by zhengbw0324** [comment]
@Snowilv 
Hello, for the sake of efficiency and simplicity, our implementation directly uses the loss value on the validation set as the metric. 
Since our testing process requires autoregressive generation, if you want to implement the calculation of metrics such as NDCG during the fine-tuning, you may need to use `Seq2SeqTrainer` and customize the `compute_metrics` function while setting `predict_with_generate=True`.
In fact, this is not worth the gain, because it will bring implementation difficulties and cause a lot of time consumption. In addition, using the loss value when training for only 3/4 epochs is actually enough.

**Comment by Snowilv** [comment]
Yeah, thanks I get it.

Besides I'm confused that when setting num_beams=20, most predictions are 20 identical items, whose 4 tokens are the same even though num_beams=20. I suppose there should be 20 various items according to the beam search.
